src/modules/drawing/components/DrawingInteractions.vue

```diff
@@ -8,28 +8,59 @@ import DrawingMarkerInteraction from '@/modules/drawing/components/DrawingMarker
 import DrawingMeasureInteraction from '@/modules/drawing/components/DrawingMeasureInteraction.vue'
 import DrawingSelectInteraction from '@/modules/drawing/components/DrawingSelectInteraction.vue'
 import DrawingTextInteraction from '@/modules/drawing/components/DrawingTextInteraction.vue'
+import ExtendLineInteraction from '@/modules/drawing/components/ExtendLineInteraction.vue'
+import ExtendMeasureInteraction from '@/modules/drawing/components/ExtendMeasureInteraction.vue'
 import { EditMode } from '@/store/modules/drawing.store'
-
+import log from '@/utils/logging'
 // DOM References
 const selectInteraction = ref(null)
 const currentInteraction = ref(null)
-
 const store = useStore()
 const currentDrawingMode = computed(() => store.state.drawing.mode)
 const editMode = computed(() => store.state.drawing.editingMode)
 
+let selectedLineFeature = null
+
 const specializedInteractionComponent = computed(() => {
+    let selectedInteraction = null
     switch (currentDrawingMode.value) {
         case EditableFeatureTypes.ANNOTATION:
-            return DrawingTextInteraction
+            selectedInteraction = DrawingTextInteraction
+            break
         case EditableFeatureTypes.LINEPOLYGON:
-            return DrawingLineInteraction
+            selectedInteraction = DrawingLineInteraction
+            break
         case EditableFeatureTypes.MARKER:
-            return DrawingMarkerInteraction
+            selectedInteraction = DrawingMarkerInteraction
+            break
         case EditableFeatureTypes.MEASURE:
-            return DrawingMeasureInteraction
+            selectedInteraction = DrawingMeasureInteraction
+            break
+    }
+    if (editMode.value === EditMode.EXTEND) {
+        const isMeasure =
+            selectedLineFeature?.get('editableFeature')?.featureType ===
+            EditableFeatureTypes.MEASURE
+        const isLine =
+            selectedLineFeature?.get('editableFeature')?.featureType ===
+            EditableFeatureTypes.LINEPOLYGON
+        if (isMeasure) {
+            selectedInteraction = ExtendMeasureInteraction
+        } else if (isLine) {
+            selectedInteraction = ExtendLineInteraction
+        } else {
+            log.error('Invalid feature type for extend mode')
+
+            selectedInteraction = null
+        }
+    }
+    // Make sure that the modify interaction is disabled when we are in draw / extend mode
+    if (selectedInteraction) {
+        selectInteraction.value?.setActive(false)
+    } else {
+        selectInteraction.value?.setActive(true)
```

Comment:
`selectInteraction.value?.setActive(!selectedInteraction)`
could do the same, no?

Reply:
yes :)